### PR TITLE
Pw 7970/express unit tests

### PR DESCRIPTION
**Pull request by zenit2001**

### CODE REVIEW

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/amazonPayExpressPart2.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      {
+        ID: 'EUR002',
+      },
+    ],
+  };
+});
+
+describe('AmazonPay Express', () => {
+  describe('AmazonPay Express Shipping methods change', () => {
+    it('Mounting the button', async () => {
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/amazonPayExpressPart2.test.js:30]: The test name is not very clear. It('Should show the updated shipping methods')

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/applePayExpress.test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    },
+  };
+});
+
+afterEach(() => {
+    jest.resetModules();
+  });
+
+describe('Apple Pay Express', () => {
+  it('Apple Pay handleError', () => {
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/applePayExpress.test.js:29]: it('Should handle Apple Pay errors'

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/applePayExpress.test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        <div id="result">Apple Pay Result</div>
+        <form id="showConfirmationForm"> Confirmation form</form>
+      `;
+    window.HTMLFormElement.prototype.submit = () => {};
+    const rejectApplePay = jest.fn();
+    handleError(rejectApplePay);
+    expect(rejectApplePay).toBeCalledTimes(1);
+  });
+
+  it('Apple Pay handleAuthorize should resolve', () => {
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/applePayExpress.test.js:40]: it('Should authorise apple pay when....')

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/applePayExpress.test.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    });
+    const resolveApplePay = jest.fn();
+    const rejectApplePay = jest.fn();
+    await callPaymentFromComponent(data, resolveApplePay, rejectApplePay);
+    expect(document.getElementById('additionalDetailsHidden').value).toBe(
+      JSON.stringify(data),
+    );
+  });
+
+  it('Apple Pay handleAuthorized', async () => {
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/__tests__/applePayExpress.test.js:87]: it('Should populate result to be sent to backend'

**Review by zaiddreakh — APPROVED** [review]

**Review by shanikantsingh — APPROVED** [review]

**Pull request merged**